### PR TITLE
Altair: Add `ProcessBlockNoVerifyAnySig`

### PR DESCRIPTION
**Pull request by terencechain**
Part of #8638 

This PR implements `ProcessBlockNoVerifyAnySig` and tests. In order to add tests, it added the following helpers:
- `ProcessAttestationsNoVerifySignature`
- `VerifyNilBeaconBlock`
- `VerifyOperationLengths`
- `BlockSignatureAltair`
    - `CalculateStateRoot`
    - `ProcessBlockForStateRoot`
- `HydrateSignedBeaconBlockAltair`
- `HydrateSBeaconBlockAltair`
- `HydrateBeaconBlockBodyAltair`

### CODE REVIEW

File: shared/testutil/helpers.go [hunk 1]
```diff
@@ -67,6 +68,42 @@ func BlockSignature(
 	return privKeys[proposerIdx].Sign(blockRoot[:]), nil
 }
 
+// BlockSignatureAltair calculates the post-state root of the block and returns the signature.
+func BlockSignatureAltair(
```
Comment (terencechain) [inline shared/testutil/helpers.go:72]: Only used for tests

File: beacon-chain/state/state-altair/field_trie_test.go [hunk 2]
```diff
@@ -1,10 +1,11 @@
-package state_altair
+package state_altair_test
```
Comment (terencechain) [inline beacon-chain/state/state-altair/field_trie_test.go:1]: Ran into annoying import cycles

File: beacon-chain/state/state-altair/field_trie.go [hunk 3]
```diff
@@ -165,41 +164,3 @@ func fieldConverters(field fieldIndex, indices []uint64, elements interface{}, c
 		return [][32]byte{}, errors.Errorf("got unsupported type of %v", reflect.TypeOf(elements).Name())
 	}
 }
-
-func handleEth1DataSlice(val []*ethpb.Eth1Data, indices []uint64, convertAll bool) ([][32]byte, error) {
```
Comment (terencechain) [inline beacon-chain/state/state-altair/field_trie.go:169]: Deleted duplicate code. Turns out we can share this with `stateV0` pkg

File: beacon-chain/core/altair/transition.go [hunk 4]
```diff
@@ -22,6 +23,82 @@ import (
 // reasonable amount of time.
 var SkipSlotCache = cache.NewSkipSlotCache()
 
+func CalculateStateRoot(
```
Comment (rauljordan) [inline beacon-chain/core/altair/transition.go:26]: Comment needed

**Review by rauljordan — APPROVED** [review]

**Pull request merged**